### ISSUE TITLE
New Module Ideas For Image Sequencer

### ISSUE THREAD

**Issue by HarshKhandeparkar** [publiclab/image-sequencer#694]
Hello everyone! Do you have any good module ideas for `image-sequencer`?

### Ideas

If you have seen or heard of image processing steps even of you don't know what they mean or how they sre different from others or if you know some great links where such reliable information can be found, then please let us know in the comments below.

If you have get ideas for modules(even original ones) and you don't know how to implement them, then you can open a new issue for it or let us know in the comments below.

If you know about some module and know how to implement it in the codebase, then please feel free to open a PR.

If you don't have any original idea then you can try implementing one of [these](https://github.com/publiclab/image-sequencer/labels/module).

****

## Thank you!

**Comment by HarshKhandeparkar** [comment]
I'm already working on a flip image module.

**Comment by HarshKhandeparkar** [comment]
Flip image module done! https://github.com/publiclab/image-sequencer/pull/696

**Comment by Mridul97** [comment]
The idea of a tint module seems interesting, I would like to work on this one! @jywarren

**Comment by HarshKhandeparkar** [comment]
@jywarren @publiclab/is-reviewers there was an old issue about a module which can remove a specific color. I have an idea about how that can be done but I'm not sure.

The module can fo through all the pixels and search for the provided color (maybe with some tolerance). If the color is found then replace it with the average of all four adjacent pixels.

@Mridul97 @VibhorCodecianGupta any thoughts? Or would you like to work on this?

The issue:
https://github.com/publiclab/image-sequencer/issues/225

**Comment by HarshKhandeparkar** [comment]
If nobody else wants to work on it the. i can try. I'll require a sample image though. @jywarren?

**Comment by Mridul97** [comment]
@HarshKhandeparkar Sounds good! would like to work on this, but I think we should wait for @jywarren thoughts about filling the pixels with average of adjacent pixels! Thanks!

**Comment by jywarren** [comment]
Hmm, interesting! Maybe we could replace with a greyscale pixel by
averaging the color channels for that pixel?

**Comment by jywarren** [comment]
I want to mention some ideas from an old issue that could be consolidated into this one, maybe?

https://github.com/publiclab/image-sequencer/issues/165

**Comment by jywarren** [comment]
And @VibhorCodecianGupta maybe check out this fontpath module here? https://github.com/publiclab/image-sequencer/issues/431

it may allow us to avoid canvas although if we can get a good pure JavaScript version of canvas running, or otherwise make it simple to support for both node and browser, we could go that way too.

**Comment by jywarren** [comment]
Ah, also - threshold!  I started here but never finished! https://github.com/publiclab/image-sequencer/pull/123

**Comment by Divy123** [comment]
@aashna27 maybe we can elaborate.
Also if you can help me with 
[https://github.com/publiclab/image-sequencer/issues/542](url)

**Comment by aashna27** [comment]
> @aashna27 maybe we can collaborate.


Yeah sure we can divide among the categories, I ll get back to you after some research.

**Comment by Harshithpabbati** [comment]
We can have these special effects
Glow 
Noise 
Color matrix 
HDR
@publiclab/is-reviewers what do you think about it???

**Comment by HarshKhandeparkar** [comment]
Can you please explain those SFX?

**Comment by HarshKhandeparkar** [comment]
@jywarren can we pin this issue so that everybody can discuss?

**Comment by HarshKhandeparkar** [comment]
Some module ideas in #165

**Comment by HarshKhandeparkar** [comment]
Temporarily unpinning this to pin the check-in

**Comment by blurry-x-face** [comment]
Unpinned the check-in to pin this issue as check-in is shifted to another repository.
